File: 2022/c/01_calories/calories.c
// elf calories
#include <stdio.h>
#include <stdlib.h>

#define SIZE 10

int main() {
    FILE* file = NULL;
    char elfCurrent[SIZE];
    int maxCal = 0, theElf = 0, elfCount = 1;

    if (!(file = fopen("input.txt", "r"))) {
        puts("FILE DID NOT OPEN");
        exit(EXIT_FAILURE);
    }

    while (fgets(elfCurrent, SIZE, file)) {
        int totalElf = 0;
        printf("\n Elf %i\n -------\n", elfCount);
        for (int i = 1; elfCurrent[0] != '\n' && !feof(file); i++) {
            printf("%2i: %s", i, elfCurrent);
            totalElf += atoi(elfCurrent);
            fgets(elfCurrent, SIZE, file);
        }
        printf("TOTAL: %i", totalElf);
        if (totalElf > maxCal) {
            maxCal = totalElf;
            theElf = elfCount;
        }
        elfCount++;
        puts("");
    }

    fclose(file);

    printf("\nelf %i has %i calories", theElf, maxCal);
}

File: 2022/go/01_calories/calories.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strconv"
)

func main() {
    elf_count, max, max_elf := 1, 0, 0

    file, err := os.Open("input")
    if err != nil {
        fmt.Fprintf(os.Stderr, "%s", err)
    }
    defer file.Close()

    buf := bufio.NewScanner(file)

    elf := ""
    for buf.Scan() {
        fmt.Printf("\n Elf %d\n -------\n", elf_count)

        elf = buf.Text()
        if elf == "" {
            continue
        }

        total := 0
        for i := 1; elf != "" && buf.Scan(); i++ {
            calories, _ := strconv.Atoi(elf)
            fmt.Printf("%2d: %d\n", i, calories)
            total += calories
            elf = buf.Text()
        }

        fmt.Printf("total: %d\n", total)
        if total > max {
            max = total
            max_elf = elf_count
        }
        elf_count++
    }

    fmt.Printf("\nElf %d has %d calories", max_elf, max)
}
